src/controls/viewPicker/ViewPicker.tsx

```diff
@@ -0,0 +1,200 @@
+        super(props);
+
+        telemetry.track('ViewPicker');
+        this.state = {
+            results: [],
+            loading: false
+        }
+
+        this.async = new Async(this);
+    }
+
+
+    public componentDidMount(): void {
+        // Start retrieving the list views
+        this.loadViews();
+    }
+
+  /**
+   * componentDidUpdate lifecycle hook
+   * @param prevProps
+   * @param prevState
+   */
+    public componentDidUpdate(prevProps: IViewPickerProps, _prevState: IViewPickerState): void {
+        if (
+            this.props.listId !== prevProps.listId || 
+            this.props.webAbsoluteUrl !== prevProps.webAbsoluteUrl ||
+            this.props.orderBy !== prevProps.orderBy 
+            ) {
+          this.loadViews();
+        }
+
+        if(prevProps.selectedView !== this.props.selectedView){
+          this.setSelectedViews();
+        }
+    }
+    
+    /**
+     * Called when the component will unmount
+    */
+    public componentWillUnmount(): void {
+        if (typeof this.async !== 'undefined') {
+            this.async.dispose();
+        }
+    }
+
+    private async loadViews(): Promise<void> {
+
+        // Show the loading indicator and disable the dropdown
+        this.setState({ loading: true });
+
+        const viewsToExclude: string[] = this.props.viewsToExclude || [];
+        let options: IDropdownOption[] = [];
+        const service: ISPService = SPServiceFactory.createService(this.props.context, true, 5000, this.props.webAbsoluteUrl);
+        let results = await service.getViews(
+          this.props.listId,
+          this.props.webAbsoluteUrl,
+          this.props.orderBy,
+          this.props.filter
+        );
+
+        // Start mapping the views that are selected
+        results.value.forEach((view: ISPView) => {
+          if (this.props.selectedView === view.Id) {
+            this.selectedKey = view.Id;
+          }
+           // Make sure that the current view is NOT in the 'viewsToExclude' array
+           if (viewsToExclude.indexOf(view.Title) === -1 && viewsToExclude.indexOf(view.Id) === -1) {
+            options.push({
+              key: view.Id,
+              text: view.Title
+            });
+          }
+        });
+
+        if (this.props.showBlankOption) {
+          // Provide empty option
+          options.unshift({
+            key: EMPTY_VIEW_KEY,
+            text: '',
+          });
+        } else{
+          // Option to unselect the view
+          options.unshift({
+            key: EMPTY_VIEW_KEY,
+            text: EMPTY_VIEW_KEY
+          });
+        }
+
+        this.setSelectedViews();
+        // Update the current component state
+        this.setState({
+          loading: false,
+          results: options
+        });
+        
+    }
+
+    /**
+     * Set the currently selected views(s);
+     */
+    private setSelectedViews(): void {
+      this._selectedView = cloneDeep(this.props.selectedView);
```

Comment:
Why do you have `selectedView` as both field and property in the state?
I don't get the point to be honest.

Reply:
This global variable is used to send the selected value to the callback function ```onSelectionChanged``` as below code did not work with state. 

```
  private onChange = (event: React.FormEvent<HTMLDivElement>, option: IDropdownOption, index?: number): void => {
    const { multiSelect, onSelectionChanged } = this.props;

    if (multiSelect) {
      let selectedViews = this.state.selectedView ? cloneDeep(this.state.selectedView) as string[] : [];

      if (option.selected) {
        selectedViews.push(option.key.toString());
      }
      else {
        selectedViews = selectedViews.filter(view => view !== option.key);
      }
      this.setState({
          selectedView : selectedViews 
      })
    }
    else {
      this.setState({
          selectedView : option.key.toString()
      })
    }
    if (onSelectionChanged) {
      onSelectionChanged(cloneDeep(this._selectedView));
    }
  }
```

Hence referred this code from ListPicker control as in below screen shot
![image](https://github.com/pnp/sp-dev-fx-controls-react/assets/47456098/adc832dc-c6e7-4ea5-add5-0665fd042d2a)

Thanks,
Nishkalank

Reply:
@NishkalankBezawada - `ListPicker` doesn't have selected lists in the state, only as a field.
Your code will work if you just pass `selectedViews` or `option.key.toString()` to the `onSelectionChanged`.
Please, leave either field or state.

Reply:
Hello @AJIXuMuK 
Your suggestion worked like a charm. Thanks for your guidance. Updated accordingly in the latest commit.

Thanks,
Nishkalank